sdk/core/Azure.Core.Experimental/src/Primitives/README.md

```diff
@@ -0,0 +1,55 @@
+# Binary Data
+## Overview
+ The `BinaryData` type provides a lightweight abstraction for a payload of bytes. This type integrates with [ObjectSerializer](../../../Azure.Core/src/ObjectSerializer.cs) to allow for serializing and deserializing payloads. It provides convenient helper methods to get out commonly used primitives, such as streams, strings, or bytes. The assumption when converting to and from string is that the encoding is UTF-8.
+ 
+ ## Data ownership
+ The ownership model of the underlying bytes varies depending on how the instance is constructed:
+ If created using the static factory method, `FromMemory(ReadOnlyMemory<byte>)`, the passed in bytes will be wrapped, rather than copied. This is useful in scenarios where performance is critical and/or ownership of the bytes is controlled completely by the consumer, thereby allowing the enforcement of whatever ownership model is needed.
+ 
+ If created using the `BinaryData(ReadOnlySpan<byte>)` constructor, `BinaryData` will maintain its own copy of the underlying bytes. This usage is geared more towards scenarios where the ownership of the bytes might be ambiguous to users of the consuming code. By making a copy of the bytes, the payload is guaranteed to be immutable. For all other constructors and static factory methods, BinaryData will assume ownership of the underlying bytes.
+
+ ## Usage
+ The main value of this type is its ability to easily convert from string to bytes to stream. This can greatly simplify API surface areas by exposing this type as opposed to numerous overloads or properties.
+ 
+ ```c#
+ var data = new BinaryData("some data");
```

Comment:
Use snippets.

Reply:
Was not able to use snippet for SB code as it is in separate directory.